soql-analyzer/src/main/scala/com/socrata/soql/analyzer2/PositionInfo.scala

```diff
@@ -2,140 +2,200 @@ package com.socrata.soql.analyzer2
+  def this(source: Source[RNS]) = this(source, source)
 
-  def logicallyReposition[RNS2 >: RNS](source: Option[ScopedResourceName[RNS2]], pos: Position) = new AtomicPositionInfo(source, pos, physicalSource, physicalPosition)
+  override def reReference[RNS2 >: RNS](newReference: Source[RNS2]) = new AtomicPositionInfo(source, newReference)
 
-  def asAtomic: AtomicPositionInfo[RNS] = this
+  override def asAtomic: AtomicPositionInfo[RNS] = this
 
-  override def toString = s"AtomicPositionInfo($logicalSource, $logicalPosition, $physicalSource, $physicalPosition)"
+  override def toString =
+    if(source == reference) {
+      s"AtomicPositionInfo($source)"
+    } else {
+      s"AtomicPositionInfo($source, $reference)"
+    }
+
+  override def hashCode = source.hashCode ^ reference.hashCode
+  override def equals(o: Any) =
+    o match {
+      case that: AtomicPositionInfo[_] =>
+        this.source == that.source &&
+          this.reference == that.reference
+      case _ => false
+    }
 }
 
+
 object AtomicPositionInfo {
-  val None = new AtomicPositionInfo[Nothing](Option.empty, NoPosition)
+  val Synthetic = new AtomicPositionInfo[Nothing](Source.Synthetic)
 
   implicit def serialize[RNS: Writable] = new Writable[AtomicPositionInfo[RNS]] {
     override def writeTo(buffer: WriteBuffer, api: AtomicPositionInfo[RNS]): Unit = {
-      if((api.logicalSource eq api.physicalSource) && (api.logicalPosition eq api.physicalPosition)) {
+      if(api.source eq api.reference) {
         // most expressions are not repositioned, so don't duplicate them when serialized if possible
         buffer.write(0)
-        buffer.write(api.logicalSource)
-        buffer.write(api.logicalPosition)
+        buffer.write(api.source)
       } else {
         buffer.write(1)
-        buffer.write(api.logicalSource)
-        buffer.write(api.logicalPosition)
-        buffer.write(api.physicalSource)
-        buffer.write(api.physicalPosition)
+        buffer.write(api.source)
+        buffer.write(api.reference)
       }
     }
   }
 
   implicit def deserialize[RNS: Readable] = new Readable[AtomicPositionInfo[RNS]] {
     override def readFrom(buffer: ReadBuffer): AtomicPositionInfo[RNS] = {
-      buffer.read[Int]() match {
-        case 0 =>
-          new AtomicPositionInfo(
-            buffer.read[Option[ScopedResourceName[RNS]]](),
-            buffer.read[Position]()
-          )
-        case 1 =>
-          new AtomicPositionInfo(
-            logicalSource = buffer.read[Option[ScopedResourceName[RNS]]](),
-            logicalPosition = buffer.read[Position](),
-            physicalSource = buffer.read[Option[ScopedResourceName[RNS]]](),
-            physicalPosition = buffer.read[Position]()
-          )
-        case other =>
-          fail("Unknown AtomicPositionInfo tag " + other)
+      buffer.version match {
+        case Version.V0 =>
```

Comment:
Since we've got internal users now, it's time to start actually using the serialization-versioning stuff to make sure deploys don't cause query outages (the error format has changed too, but those happen way less frequently and so will hopefully not actually be a problem during the brief period of inconsistency during a deploy)